src/lib/core/evio.c

```diff
@@ -344,19 +344,22 @@ int
 evio_service_bind(struct evio_service *service, const char *uri)
 {
 	struct uri u;
-	if (uri_parse(&u, uri) || u.service == NULL) {
+	if (uri_create(&u, uri) || u.service == NULL) {
 		diag_set(SocketError, sio_socketname(-1),
 			 "invalid uri for bind: %s", uri);
 		return -1;
 	}
 
-	snprintf(service->serv, sizeof(service->serv), "%.*s",
-		 (int) u.service_len, u.service);
-	if (u.host != NULL && strncmp(u.host, "*", u.host_len) != 0) {
-		snprintf(service->host, sizeof(service->host), "%.*s",
-			(int) u.host_len, u.host);
+	if (u.service != NULL) {
+		snprintf(service->serv, sizeof(service->serv), "%s",
+			 u.service);
+	}
+	if (u.host != NULL && strcmp(u.host, "*") != 0) {
+		snprintf(service->host, sizeof(service->host), "%s",
+			u.host);
```

Comment:
As you see here, there are some actions with service->host we use serv->uri instead we should make some changes with uri internals. SHould we do it?

Reply:
I don't see any actions done with host and service beside checking them for NULL so that shouldn't be a problem. However, in the future, we will probably pass an uri object to iproto_listen (because we will need to pass uri options), which means that uri will be parsed at the upper level and it will probably be the best option to pass host and service (instead of uri string or object) to evio_service_bind. This means that storing an uri object in evio_service doesn't make sense. So let's leave everything as is for now.